src/libraries/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -650,6 +650,9 @@
   <data name="InvalidJsonTypeInfoOperationForKind" xml:space="preserve">
     <value>Invalid JsonTypeInfo operation for JsonTypeInfoKind '{0}'.</value>
   </data>
+  <data name="JsonTypeInfoAlreadyBeingConfigured" xml:space="preserve">
+    <value>This JsonTypeInfo instance is already being configured by another thread.</value>
```

Comment:
why would user see this? what would they do with this error?

Reply:
See this test for a reproduction: https://github.com/dotnet/runtime/pull/82198/files#diff-d11a79a6de1f66ca1f08f73f3a3698e0eb884f6f2784a44512be662e07425753R483-R533

It would essentially involve users creating a resolver that ignores the `JsonSerializerOptions` parameter and instead returns a pre-populated instance and moreover this is being invoked by multiple threads. It's a theoretical concerns that we still need to detect and fail on. In that sense it is similar to how we intercept invalid `JsonTypeInfo` outputs already here:

https://github.com/dotnet/runtime/blob/cb30e97f8397e5f87adee13f5b4ba914cc2c0064/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs#L718-L726